imgui-core/imgui/buildfile

```diff
@@ -0,0 +1,37 @@
+intf_libs = # Interface dependencies.
+impl_libs = # Implementation dependencies.
+
+lib{imgui-core}: {cxx hxx}{*}
+
+# Build options.
+out_pfx_inc = [dir_path] $out_root/imgui/
+src_pfx_inc = [dir_path] $src_root/imgui/
+
+cxx.poptions =+ "-I$out_pfx_inc" "-I$src_pfx_inc" -DIMGUI_EXPORT
```

Comment:
> Well it is somehow complicated. I don't know why building DLLs has to be so complicated on Windows 😆

I don't think it's that complicated, and having a hard import/export symbol check helps with finding issues related to ABI.
That being said, here the issue is mainly that the upstream project didn't consider the backends as potentially separate libraries, so they didn't have separate names (defines) for stating which library will be static or shared. If that was done (the saem way default library projects are done using `bdep new -t lib`, see the export header generated), then the rest is trivial. Also personally I really don't like that on non-windows platforms the symbols are exported by default, but that's another story.

> I just mirrored what SFML does.

No worries, I understand that part, it was more a remark about the upstream code being "naive" about symbol handling (but it's designed for static first, so it's understandable).

>  If you have any suggestion to improve this handling, I will happily incorporate it.

Nope, it was just a note/remark, it's an upstream problem ^^
I'll close this so that you can focus on the rest.